src/providers/formattingProvider.utils.ts

```diff
@@ -1,6 +1,43 @@
 import * as vscode from 'vscode';
 import { FormatOptions } from './formattingProvider.types';
 
+declare global {
```

Comment:
Let's not overwrite the global.String.replaceAll function, that'd be confusing to a lot of folks and I don't think it's recommended. Instead, let's write our own function and call that instead. That way we can easily test our new function as well

Reply:
I did not overwrite it :). I add this new function to `String` type.

Reply:
Understood, I was a bit confused because EcmaScript 2021 does have [`String.prototype.replaceAll`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#specifications), though for some reason we're compiling to a lower version of ES. nbd, thanks for making the change :)